web/pandas/pdeps/0014-string-dtype.md

```diff
@@ -215,31 +225,33 @@ and NaN semantics.
+|---------------------------------------------|---------------------------------------------------------------|---------------------------------------|----------|
+| Unspecified (inference)                     | `StringDtype(storage="pyarrow"\|"python", na_value=np.nan)`   | "str"                                 | (1)      |
+| `"str"` or `StringDtype(na_value=np.nan)`   | `StringDtype(storage="pyarrow"\|"python", na_value=np.nan)`   | "str"                                 | (1)      |
+| `StringDtype("pyarrow", na_value=np.nan)`   | `StringDtype(storage="pyarrow", na_value=np.nan)`             | "str"                                 |          |
+| `StringDtype("python", na_value=np.nan)`    | `StringDtype(storage="python", na_value=np.nan)`              | "str"                                 |          |
+| `StringDtype("pyarrow")`                    | `StringDtype(storage="pyarrow", na_value=pd.NA)`              | "string[pyarrow]"                     |          |
+| `StringDtype("python")`                     | `StringDtype(storage="python", na_value=pd.NA)`               | "string[python]"                      |          |
+| `"string"` or `StringDtype()`               | `StringDtype(storage="pyarrow"\|"python", na_value=pd.NA)`    | "string[pyarrow]" or "string[python]" | (1)      |
+| `StringDtype("pyarrow_numpy")`              | `StringDtype(storage="pyarrow", na_value=np.nan)`             | "string[pyarrow_numpy]"               | (2)      |
 
 Notes:
 
 - (1) You get "pyarrow" or "python" depending on pyarrow being installed.
-- (2) Those three rows are backwards incompatible (i.e. they work now but give
-  the NA-variant), see the "Backward compatibility" section below.
-- (3) "pyarrow_numpy" is kept temporarily because this is already in a released
+- (2) "pyarrow_numpy" is kept temporarily because this is already in a released
   version, but we can deprecate it in 2.x and have it removed for 3.0.
 
-For the new default string dtype, only the `"string"` alias can be used to
-specify the dtype as a string, i.e. a way would not be provided to make the
+For the new default string dtype, only the `"str"` alias can be used to
+specify the dtype as a string, i.e. pandas would not provide a way to make the
 underlying storage (pyarrow or python) explicit through the string alias. This
-string alias is only a convenience shortcut and for most users `"string"` is
+string alias is only a convenience shortcut and for most users `"str"` is
 sufficient (they don't need to specify the storage), and the explicit
-`pd.StringDtype(...)` is still available for more fine-grained control.
+`pd.StringDtype(storage=..., na_value=np.nan)` is still available for more
+fine-grained control.
+
+Also for the existing variant using `pd.NA`, specifying the storage through the
+string alias could be deprecated, but that is left for a separate decision.
```

Comment:
Will there be an API to query the storage?  I.e., if you did `pd.Series(["abc", "def"], dtype="str")`, could you query whether the storage was `pyarrow` or `python` ?

Reply:
I explicitly mentioned that the new dtype reuses the existing `StringDtype` class, and that does have a `.storage` attribute. So no plans to change that.